ideascube/serveradmin/catalog.py

```diff
@@ -762,21 +763,26 @@ def update_cache(self):
 
         for remote in self._remotes.values():
             # TODO: Get resumable.urlretrieve to accept a file-like object?
-            fd, tmppath = tempfile.mkstemp()
-            os.close(fd)
+            with tempfile.NamedTemporaryFile() as fd:
+                tmppath = fd.name
 
-            def _progress(*args):
-                self._progress(' {}'.format(remote.name), *args)
+                def _progress(*args):
+                    self._progress(' {}'.format(remote.name), *args)
 
-            # TODO: Verify the download with sha256sum? Crypto signature?
-            urlretrieve(remote.url, tmppath, reporthook=_progress)
-
-            catalog = load_from_file(tmppath)
-
-            # TODO: Handle content which was removed from the remote source
-            self._available.update(catalog['all'])
-
-            os.unlink(tmppath)
+                # TODO: Verify the download with sha256sum? Crypto signature?
+                try:
+                    urlretrieve(remote.url, tmppath, reporthook=_progress)
+                except ConnectionError:
+                    print("Warning: Impossible to connect to the remote"
+                          " {remote.name}({remote.url}).\n"
+                          "Continue anyway without this remote."
+                          .format(remote=remote))
+                    continue
+
+                catalog = load_from_file(tmppath)
+
+                # TODO: Handle content which was removed from the remote source
+                self._available.update(catalog['all'])
```

Comment:
This whole block is harder to review than it should be, because you didn't just catch the `ConnectionError` as the commit message says, you also indented everything by using the `NamedTemporaryFile` as a context manager.

The fact you did both at once is what confused me into thinking the temporary files might never get removed, I think.

Splitting it into two commits would have been nice. :wink: